config/v1/types_oauth.go

```diff
@@ -1,37 +1,416 @@
 package v1
 
-import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	operatorsv1alpha1api "github.com/openshift/api/operator/v1alpha1"
+)
+
+///////////////////////////////////////////////
+// OAuth Server and Identity Provider Config //
+///////////////////////////////////////////////
 
 // +genclient
 // +genclient:nonNamespaced
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
 // OAuth holds cluster-wide information about OAuth.  The canonical name is `cluster`
-// TODO this object is an example of a possible grouping and is subject to change or removal
 type OAuth struct {
-	metav1.TypeMeta `json:",inline"`
-	// Standard object's metadata.
-	metav1.ObjectMeta `json:"metadata,omitempty"`
-
-	// spec holds user settable values for configuration
-	Spec OAuthSpec `json:"spec"`
-	// status holds observed values from the cluster. They may not be overridden.
-	Status OAuthStatus `json:"status"`
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              OAuthSpec   `json:"spec"`
+	Status            OAuthStatus `json:"status,omitempty"`
 }
 
 type OAuthSpec struct {
-	// options for configuring the embedded oauth server.
-	// possibly wellknown?
+	//IdentityProviders is an ordered list of ways for a user to identify themselves
+	IdentityProviders []IdentityProvider `json:"identityProviders"`
+
+	// GrantConfig describes how to handle grants
+	GrantConfig GrantConfig `json:"grantConfig"`
+
+	// TokenConfig contains options for authorization and access tokens
+	TokenConfig TokenConfig `json:"tokenConfig"`
+
+	// Templates allow you to customize pages like the login page.
+	Templates OAuthTemplates `json:"templates"`
 }
 
 type OAuthStatus struct {
+	operatorsv1alpha1api.OperatorStatus `json:",inline"`
```

Comment:
You should define your own status object similarly how we do it in other bits of config. With exact values you care about. If you don't have any yet, leave it empty for now.

Reply:
This is causing `verify` to fail because we have an import cycle